src/boot/ksdk1.1.0/devISL23415.c

```diff
@@ -49,6 +49,216 @@
 #include "gpio_pins.h"
 #include "SEGGER_RTT.h"
 #include "warp.h"
+#include "devISL23415.h"
 
+extern volatile WarpSPIDeviceState	deviceISL23415State;
+extern volatile uint32_t		gWarpSPIBaudRateKbps;
+extern volatile uint32_t		gWarpSpiTimeoutMicroseconds;
 
+/*
+ *	From device manual, Rev. B, Page 19 of 44:
+ *
+ *		"
+ *		The ISL23415 contains two volatile 8-bit registers: the Wiper
+ *		Register (WR) and the Access Control Register (ACR).
+ *
+ *		-	0x10: ACR - DEFAULT SETTING: 0x40
+ *		-	0x00: WR - DEFAULT SETTING: 0x80
+ *		"
+ */
+void		
+initISL23415(WarpSPIDeviceState volatile *  deviceStatePointer) 
+{
+	deviceStatePointer->signalType	= (
+					kWarpTypeMaskMax
+				); /* FIXME */
+	return;
+}
+
+WarpStatus	
+readDeviceRegisterISL23415(uint8_t deviceRegister, int numberOfBytes) 
+{
+	/*
+	 *	Two ISL23415 configured in Daisy Chain Configuration.
+	 */
+
+	/*
+	 *	XXX Populate the shift-out register with the read-register command,
+	 *	followed by the register to be read, followed by a zero byte FIXME.
```

Comment:
The FIXME corresponds to the fact that constant values are set without having being defined in an enum. For example deviceISL23415State.spiSourceBuffer[0] = 0b10010000; The comments next to the instructions clear the picture a bit, so if you think they are sufficient, I will remove the FIXME. Otherwise,  I happy to elaborate what the FIXME is about.